tests/components/mqtt/test_alarm_control_panel.py

```diff
@@ -338,34 +353,49 @@ async def test_publish_mqtt_no_code(
 
 
 @pytest.mark.parametrize(
-    ("hass_config", "service", "payload"),
+    ("hass_config", "service", "payload", "can_raise"),
     [
-        (DEFAULT_CONFIG_CODE, SERVICE_ALARM_ARM_HOME, "ARM_HOME"),
-        (DEFAULT_CONFIG_CODE, SERVICE_ALARM_ARM_AWAY, "ARM_AWAY"),
-        (DEFAULT_CONFIG_CODE, SERVICE_ALARM_ARM_NIGHT, "ARM_NIGHT"),
-        (DEFAULT_CONFIG_CODE, SERVICE_ALARM_ARM_VACATION, "ARM_VACATION"),
-        (DEFAULT_CONFIG_CODE, SERVICE_ALARM_ARM_CUSTOM_BYPASS, "ARM_CUSTOM_BYPASS"),
-        (DEFAULT_CONFIG_CODE, SERVICE_ALARM_DISARM, "DISARM"),
-        (DEFAULT_CONFIG_CODE, SERVICE_ALARM_TRIGGER, "TRIGGER"),
+        (DEFAULT_CONFIG_CODE, SERVICE_ALARM_ARM_HOME, "ARM_HOME", True),
+        (DEFAULT_CONFIG_CODE, SERVICE_ALARM_ARM_AWAY, "ARM_AWAY", True),
+        (DEFAULT_CONFIG_CODE, SERVICE_ALARM_ARM_NIGHT, "ARM_NIGHT", True),
+        (DEFAULT_CONFIG_CODE, SERVICE_ALARM_ARM_VACATION, "ARM_VACATION", True),
+        (
+            DEFAULT_CONFIG_CODE,
+            SERVICE_ALARM_ARM_CUSTOM_BYPASS,
+            "ARM_CUSTOM_BYPASS",
+            True,
+        ),
+        (DEFAULT_CONFIG_CODE, SERVICE_ALARM_DISARM, "DISARM", False),
+        (DEFAULT_CONFIG_CODE, SERVICE_ALARM_TRIGGER, "TRIGGER", False),
     ],
 )
 async def test_publish_mqtt_with_code(
     hass: HomeAssistant,
     mqtt_mock_entry: MqttMockHAClientGenerator,
-    service,
-    payload,
+    service: str,
+    payload: str,
+    can_raise: bool,
 ) -> None:
     """Test publishing of MQTT messages when code is configured."""
     mqtt_mock = await mqtt_mock_entry()
     call_count = mqtt_mock.async_publish.call_count
 
     # No code provided, should not publish
-    await hass.services.async_call(
-        alarm_control_panel.DOMAIN,
-        service,
-        {ATTR_ENTITY_ID: "alarm_control_panel.test"},
-        blocking=True,
-    )
+    if can_raise:
```

Comment:
Ideally this becomes a separate test or parameterized as we want to avoid branching in tests